core/src/main/java/io/aiven/klaw/service/CacheService.java

```diff
@@ -0,0 +1,158 @@
+package io.aiven.klaw.service;
+
+import java.util.*;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.*;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+@Slf4j
```

Comment:
looks like this is critical class. some description about what it does could be better to understand.